perf(linter): rearrange rules for node type analysis

camchenry opened the pull request:
Switching these rules to use a `match node.kind()` expression rather than multiple `if let` statements. Should be faster because we check `node.kind()` fewer times, plus the linter codegen recognizes this pattern automatically and can optimize skipping this rule when not needed.

+1% on all linter benchmarks.

camchenry commented on crates/oxc_linter/src/rules/unicorn/prefer_array_find.rs:
```diff
@@ -56,159 +56,166 @@ declare_oxc_lint!(
+                        .get_inner_expression()
+                        .as_member_expression()
+                        .map(|expression| expression.object().get_inner_expression())
+                    && is_filter_call(filter_call_expr)
```
this is existing code, leaving it as-is.

camc314 commented:
### Merge activity

* **Oct 16, 4:19 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Oct 16, 4:19 PM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc).
* **Oct 16, 4:23 PM UTC**: Merged by the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc).

camchenry committed:
perf(linter): rearrange rules for node type analysis (#14648)

Switching these rules to use a `match node.kind()` expression rather than multiple `if let` statements. Should be faster because we check `node.kind()` fewer times, plus the linter codegen recognizes this pattern automatically and can optimize skipping this rule when not needed.

+1% on all linter benchmarks.
